Sources/Sampler/exchange_kernel.hpp

```diff
@@ -0,0 +1,79 @@
+// Copyright 2018 The Simons Foundation, Inc. - All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef NETKET_EXCHANGE_KERNEL_HPP
+#define NETKET_EXCHANGE_KERNEL_HPP
+
+#include <Eigen/Core>
+#include <array>
+#include "Utils/messages.hpp"
+#include "Utils/random_utils.hpp"
+
+namespace netket {
+
+// Kernel generating local random exchanges
+class ExchangeKernel {
+  // number of visible units
+  const int nv_;
+
+  // clusters to do updates
+  std::vector<std::array<Index, 2>> clusters_;
+
+  std::uniform_int_distribution<Index> distcl_;
```

Comment:
Just like `std::uniform_real_distribution` `std::uniform_int_distribution` should be constructed on the fly rather than stored as a member.